searx/engines/elasticsearch.py

```diff
@@ -0,0 +1,142 @@
+        raise ValueError('unsupported query type', engine_settings['query_type'])
+
+    if index == '':
+        raise ValueError('index cannot be empty')
+
+
+def request(query, params):
+    if query_type not in _available_query_types:
+        return params
+
+    if username and password:
+        params['auth'] = HTTPBasicAuth(username, password)
+
+    params['url'] = search_url
+    params['method'] = 'GET'
+    params['data'] = dumps(_available_query_types[query_type](query))
+    params['headers']['Content-Type'] = 'application/json'
+
+    return params
+
+
+def _match_query(query):
+    """
+    The standard for full text queries.
+    searx format: "key:value" e.g. city:berlin
+    REF: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-match-query.html
+    """
+
+    try:
+        key, values = query.split(':')
+    except:
+        raise ValueError('query format must be "key:value"')
+
+    return {"query": {"match": {key: {'query': value}}}}
+
+
+def _simple_query_string_query(query):
+    """
+    Accepts query strings, but it is less strict than query_string
+    The field used can be specified in index.query.default_field in Elasticsearch.
+    REF: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-simple-query-string-query.html
+    """
+
+    return {'query': {'simple_query_string': {'query': query}}}
+
+
+def _term_query(query):
+    """
+    Accepts one term and the name of the field.
+    searx format: "key:value" e.g. city:berlin
+    REF: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-term-query.html
+    """
+
+    try:
+        key, value = query.split(':')
+    except:
+        raise ValueError('query format must be key:value')
+
+    return {'query': {'term': {key: value}}}
+
+
+def _terms_query(query):
+    """
+    Accepts multiple terms and the name of the field.
+    searx format: "key:value1,value2" e.g. city:berlin,paris
+    REF: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-terms-query.html
+    """
+
+    try:
+        key, values = query.split(':')
+    except:
+        raise ValueError('query format must be key:value1,value2')
+
+    return {'query': {'terms': {key: values.split(',')}}}
+
+
+def _custom_query(query):
+    key, value = query.split(':')
+    custom_query = custom_query_json
+    for query_key, query_value in custom_query.items():
+        if query_key == '{{KEY}}':
+            custom_query[key] = custom_query.pop(query_key)
+        if query_value == '{{VALUE}}':
+            custom_query[query_key] = value
+    return custom_query
+
+
+def response(resp):
+    results = []
+
+    resp_json = loads(resp.text)
+    if 'error' in resp_json:
+        raise Exception(resp_json['error'])
+
+    for result in resp_json['hits']['hits']:
+        r = result['_source']
+        r['template'] = 'key-value.html'
```

Comment:
If some of the field are not string, the template will raise an exception here:
https://github.com/searx/searx/blob/c03e4c86bc49d6ef4664c038066d9f1c16e7dafc/searx/templates/oscar/result_templates/key-value.html#L9

Possible fix:
`r = {key: str(value) if not key.startswith('_') else value for key, value in r.items() }`

Reply:
Fixed in commit 7a73b41